model/sharing/sharing.go

```diff
@@ -553,17 +553,31 @@ func (s *Sharing) RevokeByNotification(inst *instance.Instance) error {
 			return err
 		}
 	}
-	s.Credentials = nil
-	s.Active = false
 
-	for i, m := range s.Members {
-		if i > 0 && m.Instance != "" {
-			s.Members[i].Status = MemberStatusRevoked
+	var err error
+	for i := 0; i < 3; i++ {
+		s.Triggers = Triggers{}
+		s.Credentials = nil
+		s.Active = false
+
+		for i, m := range s.Members {
+			if i > 0 && m.Instance != "" {
```

Comment:
The idea is to not revoke the owner of the sharing?

Reply:
Yes, it's the idea.